airflow/models/connection.py

```diff
@@ -206,13 +206,16 @@ def _parse_from_uri(self, uri: str):
 
     def get_uri(self) -> str:
         """Return connection in URI format"""
-        if "_" in self.conn_type:
+        if self.conn_type and "_" in self.conn_type:
             self.log.warning(
                 "Connection schemes (type: %s) shall not contain '_' according to RFC3986.",
                 self.conn_type,
             )
 
-        uri = f"{str(self.conn_type).lower().replace('_', '-')}://"
+        if self.conn_type:
+            uri = f"{str(self.conn_type).lower().replace('_', '-')}://"
```

Comment:
Is this `str` cast still needed now this is only called when `self.conn_type` is not None?

Reply:
sounds right